pylearn2/linear/cudnn2d.py

```diff
@@ -0,0 +1,269 @@
+
+        # The calling format has to be changed
+        img = gpu_contiguous(x)
+        kerns = gpu_contiguous(self._filters)
+        shape = GpuDnnConv.get_out_shape(
+            img.shape, kerns.shape, self._border_mode, self._subsample)
+        rval = gpu_alloc_empty(*shape)
+        desc = self._desc(img.shape, kerns.shape)
+        rval = self._conv_op(img, kerns, rval, desc)
+
+        # Format the output based on the output space
+        axes = self._output_axes
+        assert len(axes) == 4
+
+        if tuple(self._output_axes) != op_axes:
+            rval = rval.dimshuffle(*[op_axes.index(ax) for ax in
+                                     self._output_axes])
+
+        return rval
+
+    def set_batch_size(self, batch_size):
+        """
+        .. todo::
+
+            WRITEME
+
+        Parameters
+        ----------
+        batch_size : TODO
+            TODO
+        """
+        self._img_shape = tuple([batch_size] + list(self._img_shape[1:]))
+
+
+def make_random_conv2D(irange,
+                       input_space,
+                       output_space,
+                       kernel_shape,
+                       batch_size=None,
+                       subsample=(1, 1),
+                       border_mode = 'valid',
+                       message = "",
+                       rng = None):
+    """
+    .. todo::
+
+        WRITEME properly
+
+    Creates a CorrMM2D with random kernels
+
+    Parameters
+    ----------
+    irange : TODO
+    input_space : TODO
+    output_space : TODO
+    kernel_shape : 2D list or tuple
+    batch_size : int, optional
+    subsample : tuple, optional
+    border_mode : string, optional
+    message : string, optional
+    rng : optional
+    """
+
+    rng = make_np_rng(rng, default_seed, which_method='uniform')
+
+    W = sharedX(rng.uniform(-irange, irange, (output_space.num_channels,
+                                              input_space.num_channels,
+                                              kernel_shape[0],
+                                              kernel_shape[1])))
+
+    return Cudnn2D(
+        filters=W,
+        batch_size=batch_size,
+        input_space=input_space,
+        output_axes=output_space.axes,
+        subsample=tuple(subsample),
+        border_mode=border_mode,
+        filters_shape=W.get_value(borrow=True).shape,
+        message=message
+    )
+
+
+def make_sparse_random_conv2D(num_nonzero,
+                              input_space,
+                              output_space,
+                              kernel_shape,
+                              batch_size,
+                              subsample=(1, 1),
+                              border_mode='valid',
+                              message="",
+                              rng=None):
+
+    """
+    .. todo::
+
+        WRITEME properly
+
+    Creates a Cudnn2D with random kernels, where the randomly initialized
+    values are sparse
+
+    Parameter
+    ---------
```

Comment:
The section should be named "Parameters".
Rather than have empty descriptions, can you put "TODO"?